grass/process.c

```diff
@@ -23,6 +23,11 @@ void proc_set_runnable(int pid) { proc_set_status(pid, PROC_RUNNABLE); }
 void proc_set_pending(int pid)  { proc_set_status(pid, PROC_PENDING_SYSCALL); }
 
 int proc_alloc() {
+    /* Student's code goes here (Preemptive Scheduler)
+     * Collect Scheduling information of proc just before it starts.
```

Comment:
I'm not sure what you mean by "Collect Scheduling information". We should discuss it.

But I agree with you that, within `proc_alloc` and `proc_free`, students need to implement something for MLFQ.